Fixes Deny x-frame-options Automattic/jetpack#21023

amustaque97 opened the pull request:
### Changes Proposed in the PR

* In this commit I have updated `diagnoseUrlError` to handle
`x-frame-options: deny` configuration
* Created a new error class for better readability

### Screenshot
![Screenshot from 2021-09-10 22-23-39](https://user-images.githubusercontent.com/21127788/132889786-c34885be-cf53-4b1c-baee-b377b0292e19.png)

### Testing Instruction
* Replace line number 11 to `"jetpack-boost-critical-css-gen": "github:amustaque97/jetpack-boost-critical-css-gen#fix\/detect-xframe-options"` https://github.com/Automattic/jetpack/blob/master/projects/plugins/boost/package.json#L11
*  Navigate to jetpack repo/directory
* Build jetpack boost plugin `jetpack build`
* Open file `jetpack/tools/docker/wordpress/wp-config.php`
* At the end of the file add this line `header('X-Frame-Options: DENY');`
*  Run `jetpack docker up`
* Open local WordPress in browser
* Open jetpack boost and enable critical CSS
* Click on `See error message`

Fixes automattic/jetpack#21023

amustaque97 committed:
Fixes Automattic/jetpack#21023

* In this commit I have updated `diagnoseUrlError` to handle
`x-frame-options: deny` configuration
* Created a new error class for better readability

thingalon commented on lib/errors.js:
```diff
@@ -126,6 +126,15 @@ class EmptyCSSError extends UrlError {
+ */
+ class XFrameDenyError extends UrlError {
+	constructor( { url } ) {
+		super( 'XFrameDenyError', { url }, `Failed to process because of x-frame-options deny configration` );
```
```suggestion
		super( 'XFrameDenyError', { url }, `Failed to load ${ url } due to the "X-Frame-Options: DENY" header` );
```

thingalon reviewed (changes requested):
Your implementation looks good. I've suggested an improvement to include the URL in the error message. I know it's not shown in Boost, but this library may get reused in other contexts, where the error text is the only useful output when something fails. :)

amustaque97 committed:
Addressed review comments

In this commit, changed error message to be more descriptive

amustaque97 commented:
Hi @thingalon, I have addressed the review comments, please take a look at PR once again. I'm attaching a screenshot after making changes.
![Screenshot 2021-09-13 21:41:21](https://user-images.githubusercontent.com/21127788/133119810-dace70c0-0e89-4795-a492-eb391bae61a2.png)

thingalon reviewed (approved):
👍  Lovely! Thank you for contributing to the Critical CSS gen library :)
